### PR TITLE
[PROD-934] Rip out all support for PodSecurityPolicies

### PR DESCRIPTION
**Pull request by okotsopoulos**
https://broadworkbench.atlassian.net/browse/PROD-934

Following successful repair of TDR Dev, Staging and Prod to disable PodSecurityPolicies for compatibility with GKE 1.25+, we'd like to fully remove PodSecurityPolicies from our Helm charts.  I broke up this PR into commits for each chart.

This should be a "no-op" for those environments since we've disabled PSPs via terra-helmfile values for them, and successfully deployed / synced.

**Follow-on actions:**

- Clean up now-unused Helm values in [developer, integration](https://github.com/broadinstitute/datarepo-helm-definitions/pull/481) environments in datarepo-helm-definitions
- Clean up now-unused Helm values in [dev](https://github.com/broadinstitute/terra-helmfile/pull/5216), [staging, prod](https://github.com/broadinstitute/terra-helmfile/pull/5222) environments in terra-helmfile

### CODE REVIEW

File: charts/datarepo-api/values.yaml [hunk 1]
```diff
@@ -44,11 +44,7 @@ serviceAccount:
   # If not set and create is true, a name is generated using the fullname template
   name:
 rbac:
-  # Specifies whether a psp should be created
   create: false
```
Comment (snf2ye) [inline charts/datarepo-api/values.yaml:47]: Can we pull the rbac.create field too?

**Review by jack-r-warren — APPROVED** [review]
LGTM!

File: charts/datarepo-api/values.yaml (hunk 1 above)
Comment (nmalfroy) [reply charts/datarepo-api/values.yaml:47]: gah bummer

**Review by nmalfroy — APPROVED** [review]
I think this should work!

**Pull request merged**